src/test/java/org/opendevstack/provision/controller/E2EProjectAPIControllerTest.java

```diff
@@ -73,40 +77,35 @@
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
-import org.springframework.boot.test.web.client.TestRestTemplate;
 import org.springframework.http.HttpMethod;
 import org.springframework.http.MediaType;
 import org.springframework.mock.web.MockHttpServletResponse;
-import org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers;
 import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.web.servlet.MockMvc;
 import org.springframework.test.web.servlet.MvcResult;
 import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
 import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
-import org.springframework.test.web.servlet.setup.MockMvcBuilders;
-import org.springframework.web.context.WebApplicationContext;
 
 /** End to end testcase with real result data - only mock is the RestClient - to feed the json */
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureMockMvc
 @ActiveProfiles({"utest", "quickstarters"})
 @DirtiesContext
 public class E2EProjectAPIControllerTest {
 
-  private static Logger e2eLogger = LoggerFactory.getLogger(E2EProjectAPIControllerTest.class);
+  private static final Logger e2eLogger =
+      LoggerFactory.getLogger(E2EProjectAPIControllerTest.class);
 
   public static final String BITBUCKET_PARAM_FILTER = "filter";
   public static final String BITBUCKET_PARAM_PERMISSION = "permission";
 
-  public static final String TEST_ADMIN_USERNAME = BasicAuthSecurityTestConfig.TEST_ADMIN_USERNAME;
-  public static final String TEST_VALID_CREDENTIAL =
-      BasicAuthSecurityTestConfig.TEST_VALID_CREDENTIAL;
-
   private RestClientMockHelper mockHelper;
 
-  private MockMvc mockMvc;
+  @Autowired private MockMvc mockMvc;
```

Comment:
Just to double check, does this also apply `SecurityMockMvcConfigurers.springSecurity()` to the `MockMvc` instance. See below deleted line 162

Reply:
I asked myself the same thing, and apparently the class annotation `AutoConfigureMockMvc` adds spring security if it finds it as a dependency in the project.

Reply:
Since Spring Security add web filters, I tested the class with `@AutoConfigureMockMvc(addFilters = false)` with the assumption that it will fail. This was the case.
Additionally I did a manual debug to verify that the security auth annotation logic is called. It looks fine to use `@AutoConfigureMockMvc`.